### PR TITLE
role and rolebinding templates

### PR DESCRIPTION
**Pull request by adamancini**

### CODE REVIEW

**Comment by adamancini** [comment]
So today I thought I would have a look at changing the object naming patterns on rolebindings, but I can't see a compelling reason to use a different pattern than `replicated-library.names.fullname`.  I think this should be ready now.

File: charts/replicated-library/values-example.yaml [hunk 1]
```diff
@@ -364,3 +372,82 @@ persistence:
...
+    # -- Labels to add to the role
+    labels: {}
+    # -- Annotations to add to the role
+    annotations: {}
+    # -- Override the name of this object. Default name if not overwritten will be releaseName-ChartName-objectName
+    fullNameOverride:
+    # -- Type of role. Must be either: ["Role", "ClusterRole"]
+    kind: Role
+    # -- Define the selectors used for aggregated ClusterRoles
+    aggregationRule: {
```
Comment (diamonwiggins) [inline charts/replicated-library/values-example.yaml:410]: This looks like the lone difference between Roles and ClusterRoles. Perhaps we should make it a bit more known that this is for one and not the other? Also I don't think this is being used in the `Role` template?

**Review by diamonwiggins — CHANGES_REQUESTED** [review]
In general everything looks good. A few comments. Would be good to see an example snippet of YAML as well in the comments.

File: charts/replicated-library/values-example.yaml (hunk 1 above)
Comment (adamancini) [reply charts/replicated-library/values-example.yaml:410]: @diamonwiggins can do, and no it's not used in the Role type

File: charts/replicated-library/templates/lib/_rolebinding.tpl [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  {{- end }}
+  {{- with (merge ($values.annotations | default dict) (include "replicated-library.annotations" $ | fromYaml)) }}
+  annotations: {{- toYaml . | nindent 4 }}
+  {{- end }}
+subjects:
+  {{- range $values.subjects }}
+  - apiGroup: ""
+    kind: ServiceAccount
+    name: {{ . }}
+    namespace: {{ $.Release.Namespace }}
```
[the remark this answers is not in the document]
Comment (adamancini) [reply charts/replicated-library/templates/lib/_rolebinding.tpl:33]: can do

File: charts/replicated-library/values-example.yaml (hunk 1 above)
Comment (adamancini) [reply charts/replicated-library/values-example.yaml:410]: I have it as a to-do to implement it for ClusterRoles but I can add some validation

**Comment by adamancini** [comment]
testing from https://github.com/adamancini/helm-minecraft-server/blob/master/values.yaml

File: charts/replicated-library/values-example.yaml (hunk 1 above)
Comment (adamancini) [reply charts/replicated-library/values-example.yaml:410]: OK, I just went ahead and added the aggregate ClusterRole template

**Review by diamonwiggins — APPROVED** [review]
awesome

File: charts/replicated-library/README_CHANGELOG.md.gotmpl [hunk 3]
```diff
@@ -12,6 +12,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### [Unreleased]
 
+### [0.12.0
```
Comment (diamonwiggins) [inline charts/replicated-library/README_CHANGELOG.md.gotmpl:15]: Missing a closing bracket here

**Pull request merged**